rc/scdoc.kak

```diff
@@ -0,0 +1,45 @@
+# https://git.sr.ht/~sircmpwn/scdoc
+# ‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾
+
+# Detection
+# ‾‾‾‾‾‾‾‾‾
+
+hook global BufCreate .+\.scd %{
+	set-option buffer filetype scdoc
+}
+
+# Initialization
+# ‾‾‾‾‾‾‾‾‾‾‾‾‾‾
+
+hook -group scdoc-highlight global WinSetOption filetype=scdoc %{
+	require-module scdoc
+
+	add-highlighter window/scdoc ref scdoc
+	hook -once -always window WinSetOption filetype=.* %{ remove-highlighter window/scdoc}
+}
+
+
+provide-module scdoc %{
+
+# Highlighters
+# ‾‾‾‾‾‾‾‾‾‾‾‾
+
+add-highlighter shared/scdoc group
+
+add-highlighter shared/scdoc/ regex '(?<!# )((?<!\\)|\\\\)\*(\n?[^\n])*?((?<!\\)|\\\\)\*' 0:+b
+add-highlighter shared/scdoc/ regex '(?<!# )((?<!\\)|\\\\)\b_((\n?[^\n])*?)((?<!\\)|\\\\)_\b' 2:+u
```

Comment:
The first underscore character isn't highlighted as being underlined, but the second one seems to be:

![2020-06-20-182604_555x45_scrot](https://user-images.githubusercontent.com/1379068/85205344-655f2300-b30a-11ea-8f23-f1f498883452.png)

Reply:
![underline-foot-dejavu](https://user-images.githubusercontent.com/37733333/85208343-0eb01400-b31f-11ea-967e-66bbb29934b0.png)
Must be your terminal/font.

Reply:
Do you like italics or underline better?

Reply:
Correct, it's my font. I thought it was the regex because I'm used to having the underscores being highlighted as `+u` as well, in which case the glitch doesn't occur. But that's fine.

The doc states [this must be underlined](https://git.sr.ht/~sircmpwn/scdoc/tree/master/scdoc.5.scd#L48), why do you ask?

Reply:
scdoc converts `_this_` to `\fIthis\fR`. The `I` stands for italics. From [man(1)][1]:

> Exact rendering may vary depending on the output device. For instance, man will usually not be able to render italics when running in a terminal, and will typically use underlined or coloured text instead.

If you look at the linked HTML man page, you will see that they use italics.

On the other hand, underline is usually closer to the actual rendering but it makes the underscores harder to see (especially with your font).

[1]: https://jlk.fjfi.cvut.cz/arch/manpages/man/man.1

Reply:
Correct, the [implementation](https://git.sr.ht/~sircmpwn/scdoc/tree/master/src/main.c#L159) doesn't follow the specification.

I don't really like this markdown-inspired paradigm when we can simply use \*bold*, /italics/ and \_underline_.

But if the implementation uses italics for underscores, then I suppose it's more logical to follow that in the plugin.